src/Integrations/Insightly/Serializers/ProjectSerializer.php

```diff
@@ -63,15 +64,22 @@ public function toInsightlyStageChange(ProjectStage $stage): array
 
     public function fromInsightlyArray(array $insightlyArray): Project
     {
-        return (new Project(
+        $project = (new Project(
             new Name($insightlyArray['PROJECT_NAME']),
             $this->pipelineStages->getProjectStageFromId($insightlyArray['STAGE_ID']),
             new ProjectStatus($insightlyArray['STATUS']),
             new Description($insightlyArray['PROJECT_DETAILS']),
-            $this->customFieldSerializer->getIntegrationType($insightlyArray['CUSTOMFIELDS']),
-            $this->customFieldSerializer->getCoupon($insightlyArray['CUSTOMFIELDS'])
+            $this->customFieldSerializer->getIntegrationType($insightlyArray['CUSTOMFIELDS'])
         ))->withId(
             new Id($insightlyArray['PROJECT_ID'])
         );
+
+        try {
+            $coupon = $this->customFieldSerializer->getCoupon($insightlyArray['CUSTOMFIELDS']);
+            $project = $project->withCoupon($coupon);
+        } catch (InvalidArgumentException $invalidArgumentException) {
```

Comment:
When would this throw an `InvalidArgumentException` exactly? Are we not obscuring any errors here by adding a catch that does not do anything?

Reply:
I wanted to keep the return value of `CustomFieldSerializer::getCoupon` not nullable. So when the `Coupon` custom field is missing the `InvalidArgumentException` will be thrown. That means that the project has no `Coupon` code, but is still a valid project.

Maybe returning a nullable `Coupon` in this case would make more sense?

Reply:
Very good suggestion! I use `InvalidArgumentException` way to often.